src/pipeline/preprocess/buildTQLQuery.ts

```diff
@@ -1,112 +1,286 @@
-import { listify } from 'radash';
-
-import { getLocalFilters } from '../../helpers';
 import type { PipelineOperation } from '../pipeline';
 
-export const buildTQLQuery: PipelineOperation = async (req) => {
-	const { schema, bqlRequest } = req;
-	if (!bqlRequest?.query) {
-		throw new Error('BQL query not parsed');
-	}
-	const { query } = bqlRequest;
-	const currentThingSchema = '$entity' in query ? query.$entity : query.$relation;
+const separator = '___';
 
-	const thingPath = currentThingSchema.defaultDBConnector.path || currentThingSchema.name;
-	if (!thingPath) {
-		throw new Error(`No thing path in ${JSON.stringify(currentThingSchema)}`);
+export const newBuildTQLQuery: PipelineOperation = async (req) => {
+	const { enrichedBqlQuery } = req;
+	if (!enrichedBqlQuery) {
+		throw new Error('BQL query not enriched');
 	}
 
-	// todo: composite Ids
-	if (!currentThingSchema.idFields) {
-		throw new Error('No id fields');
-	}
-	const [idField] = currentThingSchema.idFields;
-	const idParam = `$${thingPath}_id`;
-	let idFilter = `, has ${idField} ${idParam};`;
-	if (query.$id) {
-		if (Array.isArray(query.$id)) {
-			idFilter += ` ${idParam} like "${query.$id.join('|')}";`;
-		} else {
-			idFilter += ` ${idParam} "${query.$id}";`;
-		}
-	}
+	let tqlStr = '';
```

Comment:
This is a good example of mutable risk. It would be better to define this as a constant and build it in a single place